src/main/java/org/jenkinsci/test/acceptance/docker/Docker.java

```diff
@@ -158,6 +122,80 @@ public DockerImage build(Class<? extends DockerContainer> fixture) throws IOExce
         }
     }
 
+    //package scope for testing purposes. Ideally we should encapsulate Docker interactions so they can be mocked
+    // and call public method.
+    void copyDockerfileDirectory(Class<? extends DockerContainer> fixture, DockerFixture f, File dir)
+            throws IOException {
+        String dockerfilePath = resolveDockerfileLocation(fixture, f);
+        copyDockerfileDirectoryFromClasspath(fixture, dockerfilePath, dir);
+    }
+
+    private String resolveDockerfileLocation(Class<? extends DockerContainer> fixture, DockerFixture f) {
+        String prefix = null;
+        if(isSpecificDockerfileLocationSet(f)) {
+            prefix = f.dockerfile().replace('.', '/').replace('$', '/');
+        } else {
+            prefix = fixture.getName().replace('.', '/').replace('$', '/');
+        }
+        return prefix;
+    }
+
+    private void copyDockerfileDirectoryFromClasspath(Class<? extends DockerContainer> fixture, String dockerfileLocation, File dir) throws IOException {
+        File jar = null;
+        try {
+            jar = Which.jarFile(fixture);
+        } catch (IllegalArgumentException e) {
+            // fall through
```

Comment:
yes I completely agree but this was originally there, and since it is an improvement I didn't want to touch this as side effect. I have seen this structure in more places so we would need to create an issue for fixing some of them. Also in this case is not bad at all (maybe we could log as info) since if/else that are below this line takes care of everything. But as I said I agree with you that we need to log these lines. 

Do you think it has sense to open another issue with these improvements?